lib/beamchmark/suite/system_info.ex

```diff
@@ -37,4 +39,16 @@ defmodule Beamchmark.Suite.SystemInfo do
       _other -> :Linux
     end
   end
+
+  @spec mem(atom()) :: integer()
+  defp mem(os) do
+    {memory_str, 0} =
+      case os do
+        :macOS -> System.cmd("sysctl", ["-n", "hw.memsize"])
+        :Linux -> System.cmd("awk", ["/MemTotal/ {print $2}", "/proc/meminfo"])
+        _os -> 0
+      end
+
+    memory_str |> String.trim() |> String.to_integer()
+  end
```

Comment:
Some comments here
1. I think that we shouldn't assume that there is `awk` installed in OS. Instead let's parse this file on our own.
2. The same for `sysctl`. I am not sure about using it. Wouldn't it be possible to read some system file manually? Could we check how benchee does it?
3. We should be prepared for situation where we couldn't get MEM info and in such a case return something like `{:error, :couldnt_get_mem_info}` or `-1`. We should then change typespec for mem in stuct SystemInfo to `pos_integer() | :unknown` or `integer()` depending on what we want to return from `mem()`.

Reply:
As per https://unix.stackexchange.com/questions/82409/are-there-versions-of-unix-that-dont-have-awk-in-default-install I will assume, that all linux distributions have awk installed. I would assume as well, that all MacOS distibutions come with sysctl.

Reply:
Well, those tools can be changed e.g. https://github.com/orhun/systeroid. You can also create an alias e.g. sysctl will use systeroid under the hood

Reply:
Hmm :thinking:  Okay so let's keep this as it is